src/main/scala/timeEdit/TimeEdit.scala

```diff
@@ -0,0 +1,173 @@
+                    '|'
+                )
+                .trim
+        
+        val timeEditNormalized = normalizeTimeEditData(timeEditGrid)
+        val kaptenAllocNormalized = normalizeKaptenAllocData(kaptenAllocGrid)
+
+        val discrepancies = collection.mutable.Set[String]()
+
+        kaptenAllocNormalized.filterNot(_.rum.startsWith("Ambulans")).foreach {
+            kaEntry => 
+                timeEditNormalized.find(teEntry =>
+                    teEntry.datum == kaEntry.datum &&
+                    teEntry.start == kaEntry.start &&
+                    teEntry.del   == kaEntry.del   &&
+                    teEntry.typ   == kaEntry.typ   &&
+                    teEntry.grupp == kaEntry.grupp
+                ) match {
+                    case Some(matchedEntry) =>
+                        if (matchedEntry.rum != kaEntry.rum) then
+                            discrepancies.add(
+                                s"Saländring för ${kaEntry.datum} ${kaEntry.start} ${kaEntry.del} ${kaEntry.typ} ${kaEntry.grupp} - Ny sal: ${matchedEntry.rum}"
+                            ) 
+                    case None => 
+                        discrepancies.add(
+                            s"Saknas i TimeEdit: ${kaEntry.datum} ${kaEntry.start} ${kaEntry.del} ${kaEntry.typ} ${kaEntry.grupp} ${kaEntry.rum}"
+                        )
+                }
+        }
+
+        discrepancies.toSet
+    
+    private case class NormalizedEntry(
+        datum: String,
+        del: String,
+        typ: String,
+        grupp: String,
+        rum: String,
+        start: String
+    )
+
+    private def normalizeTimeEditData(grid: Grid): Vector[NormalizedEntry] =
+        grid.data.flatMap { row =>
+            val datum = row(grid.colIndex("datum"))
+            val start = row(grid.colIndex("start"))
+            val del = row(grid.colIndex("del"))
+            val typ = row(grid.colIndex("typ"))
+            val lokal = row(grid.colIndex("lokal"))
+            val grupp = row(grid.colIndex("grupp"))
+
+            val rooms = lokal.split(",").map(_.trim.stripPrefix("E:"))
+            val groups = grupp.split(",").map(_.trim)
+
+            rooms.zip(groups).map { case (room, group) =>
+                NormalizedEntry(datum, del, typ, group, room, start)
+            }
+        }
+
+    private def normalizeKaptenAllocData(grid: Grid): Vector[NormalizedEntry] =
+        grid.data.map { row =>
+            val datum = row(grid.colIndex("datum"))
+            val kl = row(grid.colIndex("kl"))
+            val del = row(grid.colIndex("del"))
+            val typ = row(grid.colIndex("typ"))
+            val grupp = row(grid.colIndex("grupp"))
+            val rum = row(grid.colIndex("rum"))
+
+            val adjustedTime = adjustTimeBy15Minutes(kl, -15)
+
+            NormalizedEntry(datum, del, typ, grupp, rum, adjustedTime)
+        }
+    
+    private def adjustTimeBy15Minutes(time: String, minutesDelta: Int): String =
```

Comment:
Could we just be insensitive to minutes and just check hours? I haven't checked the algorithm but it seems error-prone to fiddle with this...